From 5fbcd6b8f1635916c0a2669649d3c96312b54288 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Tue, 14 Apr 2020 11:33:23 +0200
Subject: [PATCH] Lower required version of OpenSSL to 1.1.0

SLE 15 until SP2 (so Leap < 15.2) ships 1.1.0 which won't get upgraded.
---
 src/network/configure.json             | 4 ++--
 src/network/ssl/qsslsocket_openssl.cpp | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/network/configure.json b/src/network/configure.json
index 289d84fbb4..c414e729d7 100644
--- a/src/network/configure.json
+++ b/src/network/configure.json
@@ -61,8 +61,8 @@
             "export": "openssl",
             "test": {
                 "tail": [
-                    "#if !defined(OPENSSL_VERSION_NUMBER) || OPENSSL_VERSION_NUMBER-0 < 0x10101000L",
-                    "#  error OpenSSL >= 1.1.1 is required",
+                    "#if !defined(OPENSSL_VERSION_NUMBER) || OPENSSL_VERSION_NUMBER-0 < 0x10100000L",
+                    "#  error OpenSSL >= 1.1.0 is required",
                     "#endif",
                     "#if !defined(OPENSSL_NO_EC) && !defined(SSL_CTRL_SET_CURVES)",
                     "#  error OpenSSL was reported as >= 1.1.1 but is missing required features, possibly it's libressl which is unsupported",
diff --git a/src/network/ssl/qsslsocket_openssl.cpp b/src/network/ssl/qsslsocket_openssl.cpp
index 9b28d52e21..daddea1feb 100644
--- a/src/network/ssl/qsslsocket_openssl.cpp
+++ b/src/network/ssl/qsslsocket_openssl.cpp
@@ -1943,8 +1943,8 @@ bool QSslSocketPrivate::ensureLibraryLoaded()
         if (q_OPENSSL_init_ssl(0, nullptr) != 1)
             return false;
 
-        if (q_OpenSSL_version_num() < 0x10101000L) {
-            qCWarning(lcSsl, "QSslSocket: OpenSSL >= 1.1.1 is required; %s was found instead", q_OpenSSL_version(OPENSSL_VERSION));
+        if (q_OpenSSL_version_num() < 0x10100000L) {
+            qCWarning(lcSsl, "QSslSocket: OpenSSL >= 1.1.0 is required; %s was found instead", q_OpenSSL_version(OPENSSL_VERSION));
             return false;
         }
 
-- 
2.25.1

